cub/cub/detail/choose_offset.cuh

```diff
@@ -65,6 +65,22 @@ struct ChooseOffsetT
     typename std::conditional<sizeof(NumItemsT) <= 4, std::uint32_t, unsigned long long>::type;
 };
 
+/**
+ * PromoteSmallOffsetT checks NumItemsT, the type of the num_items parameter, and
+ * promotes any integral type smaller than 32 bits to a signed 32-bit integer type.
+ */
+template <typename NumItemsT>
+struct PromoteSmallOffsetT
```

Comment:
suggestion: snake case in detail namespace

Reply:
Agreed. Switched to `promote_small_offset<T>::type` and introduced an alias template of `promote_small_offset_t<T>`. Planning to do the same for `ChooseOffsetT` in a separate PR.